nuxeo-core/nuxeo-core-test/src/main/java/org/nuxeo/ecm/core/test/WorkManagerConfiguration.java

```diff
@@ -0,0 +1,172 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *
+ */
+package org.nuxeo.ecm.core.test;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.lib.stream.log.kafka.KafkaUtils;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.RuntimeFeature;
+import org.osgi.framework.Bundle;
+
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Description of the specific capabilities of a WorkManager for tests, and helper methods.
+ *
+ * @since 10.3
+ */
+public class WorkManagerConfiguration {
+
+    private static final Log log = LogFactory.getLog(WorkManagerConfiguration.class);
+
+    public static final String WORK_MANAGER_PROPERTY = "nuxeo.test.workmanager.implementation";
+
+    public static final String WORK_MANAGER_DEFAULT = "default";
+
+    public static final String WORK_MANAGER_CHRONICLE_STREAM = "chronicle-stream";
+
+    public static final String WORK_MANAGER_KAFKA_STREAM = "kafka-stream";
+
+    private static final String STREAM_WORK_MANAGER_STORESTATE_ENABLED_PROPERTY =
+            "nuxeo.test.stream.workmanager.storestate.enabled";
+
+    private static final String STREAM_WORK_MANAGER_STORESTATE_ENABLED_DEFAULT = "false";
+
+    private String workManagerType;
+
+    private boolean streamWorkManagerStoreStateEnabled;
+
+    final CoreFeature feature;
+
+    public WorkManagerConfiguration(CoreFeature feature) {
+        this.feature = feature;
+    }
+
+    protected void init() {
+        workManagerType = calcWorkManagerType();
+        streamWorkManagerStoreStateEnabled = Boolean.parseBoolean(
+                defaultProperty(STREAM_WORK_MANAGER_STORESTATE_ENABLED_PROPERTY,
+                        STREAM_WORK_MANAGER_STORESTATE_ENABLED_DEFAULT));
+        if (WORK_MANAGER_KAFKA_STREAM.equals(workManagerType)) {
+            initKafka();
+        }
+    }
+
+    protected String calcWorkManagerType() {
+        final String value = System.getProperty(WORK_MANAGER_PROPERTY);
+        if (StringUtils.isNotEmpty(value)) {
+            if (Stream.of(WORK_MANAGER_DEFAULT, WORK_MANAGER_CHRONICLE_STREAM, WORK_MANAGER_KAFKA_STREAM).noneMatch(
+                    s -> s.equals(value))) {
```

Comment:
No need for streams:

    if (!Arrays.asList(WORK_MANAGER_DEFAULT, WORK_MANAGER_CHRONICLE_STREAM, WORK_MANAGER_KAFKA_STREAM).contains(value))

Reply:
Agreed